### PR TITLE
Splash Screen Implementation

### LINKED ISSUE #48: [1pts] Opening Screen
**Fill in the blanks**:
As a developer, I would like to broadcast our logo, so that professionalism is increased.

**Describe the solution you'd like:**
- [x] center logo on this screen
- [x] trigger this screen after loading backend completes (from loading screen)
- [x] connect to Home Screen

**Definition of Done - Check as you go!**
- [x] Tasks have been implemented
- [x] Documented
- [x] Passing Tests for each task
- [x] Pull Request made
- [x] Code Reviewed

### PR DESCRIPTION
**Pull request by oreorz**
## Description:

Implements the splash screen for both iOS and Android using the [flutter_native_splash](https://pub.dev/packages/flutter_native_splash) library.

## Screenshots

splash_screenshot

### CODE REVIEW

**Comment by oreorz** [comment]
One of bullet points which haven't been addressed yet, hope to speak about it in an upcoming meeting.

**Comment by brendacano** [comment]
If you look at the builds, it failed the Flutter one and it says:

"18s
Run dart analyze
Analyzing quack_app...

warning - analysis_options.yaml:10:10 - The include file 'package:flutter_lints/flutter.yaml' in '/Users/runner/work/quack/quack/quack_app/analysis_options.yaml' can't be found when analyzing '/Users/runner/work/quack/quack/quack_app'. - include_file_not_found

1 issue found.
Error: Process completed with exit code 2."

If you could look into this and fix this issue? It seems like you just have something in your files that isn't supposed to be there maybe?

File: quack_app/test/screens/splash_screen_test.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+];
+
+bool checkPath(String path) {
+  return File(path).existsSync();
+}
+
+void main() {
+  test('All splash screen asset files exist', () {
+    for(var assetPath in assetPaths) {
+      allLoaded = checkPath(assetPath);
```
Comment (RafaelPiloto10) [inline quack_app/test/screens/splash_screen_test.dart]: Careful, this actually won't catch any assets that don't load if the last one does, ex:

Beginning <- load, therefore `= true`
Middle <- don't load, therefore `= false`
End <- load, therefore `= true`
`allLoaded = true`, therefore tests pass but should fail

File: quack_app/test/screens/splash_screen_test.dart [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  'ios/Runner/Assets.xcassets/LaunchImage.imageset/LaunchImage@2x.png',
+  'ios/Runner/Assets.xcassets/LaunchImage.imageset/LaunchImage@3x.png'
+];
+
+bool checkPath(String path) {
+  return File(path).existsSync();
+}
+
+void main() {
+  test('All splash screen asset files exist', () {
```
Comment (RafaelPiloto10) [inline quack_app/test/screens/splash_screen_test.dart]: Can you use a group? And then one way to get around the bug mentioned below:
test each file name independently and expect it to be there. For each test, name it the file path so we know which files are missing on failure

**Review by RafaelPiloto10 — CHANGES_REQUESTED** [review]

File: quack_app/test/screens/splash_screen_test.dart (hunk 1 above)
Comment (DDVD233) [reply quack_app/test/screens/splash_screen_test.dart]: Right. You can use something like:
```
for(var assetPath in assetPaths) {
      expect(checkPath(assetPath), true);
}
```

**Review by DDVD233 — APPROVED** [review]
All good (except the one Rafael mentioned)

File: quack_app/test/screens/splash_screen_test.dart (hunk 1 above)
Comment (RafaelPiloto10) [reply quack_app/test/screens/splash_screen_test.dart]: Preferably:
```dart
group("splash test" () {
      for(var assetPath in assetPaths) {
            test(assetPath, () {
                   expect(checkPath(assetPath), true);     
            }
      }
});
```

**Review by RafaelPiloto10 — APPROVED** [review]

**Pull request merged**